src/main/java/net/minecraft/entity/player/PlayerEntity.java

```diff
@@ -2615,7 +2619,9 @@ public CooldownTracker getCooldownTracker()
 
     protected float getSpeedFactor()
     {
-        return !this.abilities.isFlying && !this.isElytraFlying() ? super.getSpeedFactor() : 1.0F;
+        Module noSlow = Client.getInstance().moduleManager.getModuleByClass(NoSlow.class);
+        return !this.abilities.isFlying && !this.isElytraFlying() && !(this instanceof ClientPlayerEntity && noSlow.isEnabled2() && noSlow.getBooleanValueFromSettingName("Blocks"))
+                ? super.getSpeedFactor() : 1.0F;
```

Comment:
```suggestion
        // MODIFICATION BEGIN: also use full speed factor if NoSlow is enabled, this is a ClientPlayerEntity, and the Blocks setting is toggled
        Module noSlow = Client.getInstance().moduleManager.getModuleByClass(NoSlow.class);
        return !this.abilities.isFlying && !this.isElytraFlying() && !(this instanceof ClientPlayerEntity && noSlow.isEnabled2() && noSlow.getBooleanValueFromSettingName("Blocks"))
                ? super.getSpeedFactor() : 1.0F;
        // MODIFICATION END
```
also the summaries are probably useless I just do them lol